### PR TITLE
Improve stability of hostnames test

### LINKED ISSUE #952: [Test failure] Slow runners show issue with hostnames
```
*** [err]: Verify the nodes configured with prefer hostname only show hostname for new nodes in tests/unit/cluster/hostnames.tcl
```

https://github.com/valkey-io/valkey/actions/runs/10540936135/job/29259761374?pr=923#step:5:7524

### PR DESCRIPTION
**Pull request by madolson**
Maybe partially resolves https://github.com/valkey-io/valkey/issues/952.

The hostnames test relies on an assumption that node zero and node six don't communicate with each other to test a bunch of behavior in the handshake stake. This was done by previously dropping all meet packets, however it seems like there was some case where node zero was sending a single pong message to node 6, which was partially initializing the state.

I couldn't track down why this happened, but I adjusted the test to simply pause node zero which also correctly emulates the state we want to be in since we're just testing state on node 6, and removes the chance of errant messages. The test was failing about 5% of the time locally, and I wasn't able to reproduce a failure with this new configuration.

### CODE REVIEW

**Review by enjoy-binbin — APPROVED** [review]
nice!

File: tests/unit/cluster/hostnames.tcl [hunk 1]
```diff
@@ -174,12 +174,12 @@ test "Verify the nodes configured with prefer hostname only show hostname for ne
...
     # Also make sure we know about the isolated master, we 
     # just can't reach it.
+    resume_process [srv 0 pid]
     set master_id [R 0 CLUSTER MYID]
     assert_match "*$master_id*" [R 6 CLUSTER NODES]
 
     # Stop dropping cluster packets, and make sure everything
     # stabilizes
-    R 0 DEBUG DROP-CLUSTER-PACKET-FILTER -1
     R 6 DEBUG DROP-CLUSTER-PACKET-FILTER -1
```
Comment (zuiderkwast) [inline tests/unit/cluster/hostnames.tcl]: We shoudn't unpause node 0 before this check: "Also make sure we know about the isolated master, we just can't reach it."

You unpaused it here just to get MYID of node 0, right? We can fetch `R 0 CLUSTER MYID` early in the test case, before we pause node 0, instead.

**Review by zuiderkwast — COMMENTED** [review]
Yeah, pause process seems more robust. Good idea!

File: tests/unit/cluster/hostnames.tcl (hunk 1 above)
Comment (madolson) [reply tests/unit/cluster/hostnames.tcl]: Sure

File: tests/unit/cluster/hostnames.tcl [hunk 2]
```diff
@@ -132,11 +132,14 @@ test "Verify the nodes configured with prefer hostname only show hostname for ne
         R $j config set cluster-announce-hostname "shard-$j.com"
     }
 
+    # Grab the ID so we have it for later for validation
```
Comment (madolson) [inline tests/unit/cluster/hostnames.tcl]:
```suggestion
    # Grab the ID so we have it later for validation
```

**Review by zuiderkwast — APPROVED** [review]

**Pull request merged**